app/lib/domains.server.ts

```diff
@@ -0,0 +1,148 @@
+import * as dns from './dns.server';
+import { RecordStatus } from '@prisma/client';
+import { prisma } from '~/db.server';
+import type { RecordType } from './dns.server';
+
+interface DNSProp {
+  type: RecordType;
+  name: string;
+  value: string;
+}
+
+interface DBProp {
+  id?: number;
+  username?: string;
+  description?: string;
+  course?: string;
+  ports?: string;
+  expiresAt?: Date;
+  status?: RecordStatus;
+}
+
+export const createUserDomain = async (dnsProp: DNSProp, dbProp: DBProp) => {
```

Comment:
You could combine these two together using an [Intersection Type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types):

```ts
export const createUserDomain = async (data: Record & DnsRecord) => {
```

Now `data` will have to satisfy both types, meaning that you have to include everything in one object.  This might make it easier to pass the data down.

Reply:
Same comment for the rest of this file.

Reply:
oh, this is like a magic! Thank you